### PR TITLE
feat(daterange): allow clear on selected daterange

### PR DESCRIPTION
**Pull request by Souravpakhira**
**What** - Added enableClear prop as optional and handleClear function to clear the state.
**Why** - If we want to clear the selected date range. enableClear prop will help to clear the selection.
**How** - Added enableClear prop if its true than handleClear function will set the state to null.
**Testing** - set enableClear prop to true and than select daterange. Click on the cancel button to clear the selection.

### CODE REVIEW

**Comment by Souravpakhira** [comment]
#321

File: src/components/input-elements/DateRangePicker/DateRangePicker.tsx [hunk 1]
```diff
@@ -50,6 +50,7 @@ export interface DateRangePickerProps
   disabled?: boolean;
   color?: Color;
   locale?: Locale;
+  allowClear?: boolean;
```
Comment (mitrotasios) [inline src/components/input-elements/DateRangePicker/DateRangePicker.tsx]: lets call this `enableClear` 🙂

**Review by mitrotasios — CHANGES_REQUESTED** [review]
Really great job, thanks! 🎉

Just have a minor suggestion :)

File: src/components/input-elements/DateRangePicker/DateRangePicker.tsx [hunk 2]
```diff
@@ -67,6 +68,7 @@ const DateRangePicker = React.forwardRef<HTMLDivElement, DateRangePickerProps>((
     color = BaseColors.Blue,
     enableYearPagination = false,
     locale = enUS,
+    allowClear = false,
```
Comment (mitrotasios) [inline src/components/input-elements/DateRangePicker/DateRangePicker.tsx]: Oh and lets set this to `true` by default

**Review by mitrotasios — CHANGES_REQUESTED** [review]

File: src/components/input-elements/DateRangePicker/DateRangePicker.tsx (hunk 1 above)
Comment (Souravpakhira) [reply src/components/input-elements/DateRangePicker/DateRangePicker.tsx]: Done as per the comment

File: src/components/input-elements/DateRangePicker/DateRangePicker.tsx (hunk 2 above)
Comment (Souravpakhira) [reply src/components/input-elements/DateRangePicker/DateRangePicker.tsx]: Done as per the suggestion

**Review by mitrotasios — APPROVED** [review]

**Comment by mitrotasios** [comment]
Great Job 🥳

**Comment by Souravpakhira** [comment]
> Great Job partying_face

Thanks excited to contribute more to this project

**Pull request merged**